engine/src/main/java/org/terasology/rendering/opengl/fbms/DisplayResolutionDependentFBOs.java

```diff
@@ -90,32 +95,30 @@ private void updateFullScale() {
     public void update() {
         updateFullScale();
 
-        FBO sceneOpaqueFbo = get(READONLY_GBUFFER);
-        if (sceneOpaqueFbo.dimensions().areDifferentFrom(fullScale)) {
-            disposeAllFBOs();
-            createFBOs();
+        FBO readOnlyGBuffer = gBufferPair.getReadFbo();
+        if (readOnlyGBuffer.dimensions().areDifferentFrom(fullScale)) {
+            regenerateFbos();
             notifySubscribers();
         }
     }
 
-    private void disposeAllFBOs() {
-        for (ResourceUrn urn : fboConfigs.keySet()) {
-            fboLookup.get(urn).dispose();
+    private void regenerateFbos() {
+        for (SimpleUri urn : fboConfigs.keySet()) {
+            FBOConfig fboConfig = getFboConfig(urn);
+            fboConfig.setDimensions(fullScale.multiplyBy(fboConfig.getScale()));
+            FBO.recreate(get(urn), getFboConfig(urn));
         }
-        fboLookup.clear();
-    }
+   }
 
-    private void createFBOs() {
-        for (FBOConfig fboConfig : fboConfigs.values()) {
-            generateWithDimensions(fboConfig, fullScale.multiplyBy(fboConfig.getScale()));
+    private void disposeAllFbos() {
+        // TODO: This should be public, and should be called while disposing the DRDFboM to prevent leaks.
```

Comment:
Mostly because I wanted to discuss the idea with you, and because I want to do this (and all other disposal tweaks) in one large PR - since I'm sure there are a lot of leaks :)